### PR TITLE
make pl:s accessible

### PR DESCRIPTION
**Pull request by HookedBehemoth**

### CODE REVIEW

File: nx/source/services/pl.c [hunk 1]
```diff
@@ -6,18 +6,27 @@
...
 static SharedMemory g_plSharedmem;
 
 static Result _plGetSharedMemoryNativeHandle(Handle* handle_out);
 
-NX_GENERATE_SERVICE_GUARD(pl);
+NX_GENERATE_SERVICE_GUARD_PARAMS(pl, (PlServiceType service_type), (service_type));
 
-Result _plInitialize(void) {
+Result _plInitialize(PlServiceType service_type) {
     Result rc=0;
```
Comment (yellows8) [inline nx/source/services/pl.c:18]: `Result rc = MAKERESULT(Module_Libnx, LibnxError_BadInput);`

**Comment by yellows8** [comment]
Can you PR the required changes for this to switch-examples/nx-hbmenu too?

**Comment by HookedBehemoth** [comment]
Any information I should add to the example?

**Comment by yellows8** [comment]
no

**Pull request merged**